Extensions/XMPPMUCSUB/XMPPMUCSUB.m

```diff
@@ -487,14 +490,18 @@ - (void)handleDiscoverFeaturesForRoomQueryIQ:(XMPPIQ *)iq withInfo:(XMPPBasicTra
                                                  code:[errorElem attributeIntegerValueForName:kXMPPCode
                                                                              withDefaultValue:0]
                                              userInfo:dict];
-            [self->multicastDelegate xmppMUCSUB:self failedToDiscoverFeaturesForRoomJID:roomJID withError:error];
+            if (roomJID) {
+                [self->multicastDelegate xmppMUCSUB:self failedToDiscoverFeaturesForRoomJID:roomJID withError:error];
+            }
             return;
         }
         
         NSXMLElement *query = [iq elementForName:kXMPPQuery xmlns:XMPPMucSubDiscoInfo];
         
         NSArray<NSXMLElement *> *features = [query elementsForName:kXMPPFeature];
-        [self->multicastDelegate xmppMUCSUB:self didDiscoverFeatures:features forRoomJID:roomJID];
+        if (roomJID) {
+            [self->multicastDelegate xmppMUCSUB:self didDiscoverFeatures:features forRoomJID:roomJID];
+        }
         self->hasRequestedFeaturesForRoom[roomJID.bare] = @NO; // Set this back to NO to allow for future requests
```

Comment:
If `roomJID` is nil, won't this just blow up on line 505? Or will it just overwrite the 0 element in `hasRequestedFeaturesForRoom` ?

Also - all the rest of the changes below are silent - log the failures, or maybe these are normal cases we don't need to log/know about?

Reply:
I changed the code to send error message to all delegates